sentry-android-okhttp/src/main/java/io/sentry/android/okhttp/SentryOkHttpEvent.kt

```diff
@@ -0,0 +1,142 @@
+package io.sentry.android.okhttp
+
+import io.sentry.Breadcrumb
+import io.sentry.Hint
+import io.sentry.IHub
+import io.sentry.ISpan
+import io.sentry.SpanStatus
+import io.sentry.TypeCheckHint
+import io.sentry.android.okhttp.SentryOkHttpEventListener.Companion.CONNECTION_EVENT
+import io.sentry.android.okhttp.SentryOkHttpEventListener.Companion.CONNECT_EVENT
+import io.sentry.android.okhttp.SentryOkHttpEventListener.Companion.REQUEST_BODY_EVENT
+import io.sentry.android.okhttp.SentryOkHttpEventListener.Companion.REQUEST_HEADERS_EVENT
+import io.sentry.android.okhttp.SentryOkHttpEventListener.Companion.RESPONSE_BODY_EVENT
+import io.sentry.android.okhttp.SentryOkHttpEventListener.Companion.RESPONSE_HEADERS_EVENT
+import io.sentry.android.okhttp.SentryOkHttpEventListener.Companion.SECURE_CONNECT_EVENT
+import io.sentry.util.UrlUtils
+import okhttp3.Request
+import okhttp3.Response
+
+internal class SentryOkHttpEvent(private val hub: IHub, private val request: Request) {
+    private val eventSpans: MutableMap<String, ISpan> = HashMap()
+    private val breadcrumb: Breadcrumb
+    internal val callRootSpan: ISpan?
+    private var response: Response? = null
+
+    init {
+        val urlDetails = UrlUtils.parse(request.url.toString())
+        val url = urlDetails.urlOrFallback
+        val host: String = request.url.host
+        val encodedPath: String = request.url.encodedPath
+        val method: String = request.method
+
+        // We start the call span that will contain all the others
+        callRootSpan = hub.span?.startChild("http.client", "$method $url")
+
+        urlDetails.applyToSpan(callRootSpan)
+
+        // We setup a breadcrumb with all meaningful data
+        breadcrumb = Breadcrumb.http(url, method)
+        breadcrumb.setData("url", url)
+        breadcrumb.setData("host", host)
+        breadcrumb.setData("path", encodedPath)
+        breadcrumb.setData("method", method)
+        breadcrumb.setData("success", true)
```

Comment:
https://develop.sentry.dev/sdk/event-payloads/breadcrumbs/
`status_code` already implies the `success`